### DISCUSSION TITLE
Why Webpack?

### DISCUSSION THREAD

**Discussion by a7medev** [callstack/repack#144] [Q&A] [answered]
Webpack has been there for a long time and it's really great, but why not choose something faster like Snowpack, ESBuild, SWC or Rollup? We really need faster builds in React Native.

**Comment by zamotany** [comment, marked as answer]
1. Because it's not possible to create runnable bundle using some of the tools you mentioned. React Native is a non-standard target, which they don't support and there's a lot of customisations needed to make it work, which those tools don't support or don't allow. SWC doesn't even have stable bundler.
2. None of them support Module Federation, only Webpack does.

So TL;DR: because it's possible at all with Webpack and it has features we want to bring to React Native ecosystem.
